### PR TITLE
Added Istio SessionMigration tests to Release 3.3 and created util methods

### PR DESCRIPTION
**Pull request by hzhao-github**
Jenkins jobs:

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8863
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8865

With latest changes

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8883
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8885

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8913

### CODE REVIEW

File: integration-tests/src/test/resources/istio/istio-sessmigr-template.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2021, Oracle and/or its affiliates.
```
Comment (xiancao) [inline integration-tests/src/test/resources/istio/istio-sessmigr-template.yaml:1]: change this copyright to:
// Copyright (c) 2022, Oracle and/or its affiliates.

**Review by xiancao — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioGatewaySessionMigration.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
+// Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioGatewaySessionMigration.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigration.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigration.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigration.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+    // Generate the model.sessmigr.yaml file at RESULTS_ROOT
+    String destSessionMigrYamlFile =
+        generateNewModelFileWithUpdatedDomainUid(domainUid, "ItIstioSessionMigration", getOrigModelFile());
+
+    List<String> appList = new ArrayList();
+    appList.add(SESSMIGR_APP_NAME);
+
+    // build the model file list
+    //final List<String> modelList = Collections.singletonList(MODEL_DIR + "/" + SESSMIGR_MODEL_FILE);
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioSessionMigration.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 5]
```diff
@@ -409,7 +411,10 @@ void testSideCarBasicFunctionality() throws Exception {
 
     // create and verify one cluster mii domain
     logger.info("Create domain and verify that it's running");
-    String miiImage1 = createAndVerifyMiiImage(MODEL_DIR + "/model.sessmigr.yaml");
+    String destYamlFile = generateNewModelFileWithUpdatedDomainUid(domain7Uid,
+        "ItMonitoringExporter", getOrigModelFile());
+    String miiImage1 = createAndVerifyMiiImage(destYamlFile);
+    //String miiImage1 = createAndVerifyMiiImage(MODEL_DIR + "/model.sessmigr.yaml");
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 6]
```diff
@@ -431,7 +436,8 @@ void testSideCarBasicFunctionality() throws Exception {
     changeMonitoringExporterSideCarConfig(RESOURCE_DIR + "/exporter/rest_domainqualtrue.yaml",
         domain7Uid, domain7Namespace,
         "domainQualifier", "wls_servlet_executionTimeAverage%7Bapp%3D%22myear%22%7D%5B15s%5D",
-        "\"domain\":\"wls-sessmigr-domain-1\"");
+        "\"domain\":\"" + domain7Uid + "\"");
+    //"\"domain\":\"wls-sessmigr-domain-1\"");
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: removed

**Review by anpanigr — APPROVED** [review]

**Review by xiancao — APPROVED** [review]

**Pull request merged**